### PR TITLE
PROD-13615 - Log warning for errors with only warn flag

### PR DESCRIPTION
**Pull request by Dimitris-Ilias**
https://workable.atlassian.net/browse/PROD-13615

### CODE REVIEW

**Comment by Dimitris-Ilias** [comment]
@nikostoulas Wdyt? Can't we can do sth like this to hide indeed prediction/recommendation errors?
Didn't want to use the blacklist because this could potentially hide errors of other job boards

File: src/initializers/koa/error-handler.ts [hunk 1]
```diff
@@ -9,6 +9,8 @@ export const isBlacklisted = (err: { status: number } = {} as any, config) =>
   // tslint:disable-next-line:triple-equals
   err.status && config.blacklistedErrorCodes.some(b => err.status == b);
 
+const onlyWarn = (err: { onlyWarn: boolean }) => !!err?.onlyWarn
```
Comment (nikostoulas) [inline src/initializers/koa/error-handler.ts:12]:
```suggestion
const logLevel = (err: { logLevel: LogLevels }) => !!err?.logLevel
```

**Review by nikostoulas — COMMENTED** [review]
Yeah we could do sth like this @Dimitris-Ilias

File: src/initializers/koa/error-handler.ts (hunk 1 above)
Comment (Dimitris-Ilias) [reply src/initializers/koa/error-handler.ts]: @nikostoulas Can you take a look at the latest implementation?

I was checking https://www.npmjs.com/package/http-errors documentation
image


So based on the `properties` field, in order to make this work with `ctx.assert` maybe only thing that's needed is to use it like this on the postings server side?
`ctx.assert(token, 401, { message: calculateMessage(), logLevel: 'WARN' });`

File: src/initializers/koa/error-handler.ts [hunk 2]
```diff
@@ -33,10 +41,10 @@ export default (config, orkaOptions: Partial<OrkaOptions>) =>
...
 
       const errorArgs = (await orkaOptions.errorHandler(ctx, err, orkaOptions)) || [err as Error];
-      if (!isBlacklisted(err, config)) {
-        logger.error(...errorArgs);
-      } else {
+      if (isBlacklisted(err, config) || onlyWarn(err)) {
         logger.warn(...errorArgs);
+      } else {
+        logger.error(...errorArgs);
       }
```
Comment (nikostoulas) [inline src/initializers/koa/error-handler.ts:48]:
```suggestion
   logger[getLogLevel(err, config)](...errorArgs)
   
   and    getLoglevel sth like
   
   - if err.logLevel is given and is valid use this
   - else if blacklisted warn
   - else error
   
```

File: src/initializers/koa/error-handler.ts (hunk 1 above)
Comment (nikostoulas) [reply src/initializers/koa/error-handler.ts]: Nice interface @Dimitris-Ilias .
Left a comment for the implementation details

File: src/initializers/koa/error-handler.ts (hunk 2 above)
Comment (Dimitris-Ilias) [reply src/initializers/koa/error-handler.ts]: @nikostoulas Done, cytal?

File: src/initializers/koa/error-handler.ts [hunk 3]
```diff
@@ -2,13 +2,21 @@ import * as Koa from 'koa';
 import { getLogger } from '../log4js';
 import { OrkaOptions } from 'orka/typings/orka';
 import { omit } from 'lodash';
-
+const Levels = require('log4js/lib/levels');
```
Comment (nikostoulas) [inline src/initializers/koa/error-handler.ts:5]: 🔧 can we convert this to import?

**Review by nikostoulas — APPROVED** [review]
Nice @Dimitris-Ilias. Since we are touching error handler can we document it's behavior here: https://workable.github.io/orka/middleware.html ?

File: src/initializers/koa/error-handler.ts (hunk 3 above)
Comment (Dimitris-Ilias) [reply src/initializers/koa/error-handler.ts:5]: Done

**Pull request merged**